### ISSUE TITLE
Support dynamically loading modules with own `config.yaml`s

### ISSUE THREAD

**Issue by sdruskat** [snakemake/snakemake#2002]
*Disclaimer:* I'm new to workflows and Snakemake and don't know if this is really a good idea, so I won't be angry when someone says it isn't, and the request gets closed.

**Is your feature request related to a problem? Please describe.**

I'd like to be able to dynamically load modules, e.g., with a `for` loop such as the one below. Each module should be able to bring its own `config.yaml` file that should be honoured. This would enable me to dynamically develop a workflow with a mechanism similar to Python extension points in `pyproject.toml`.

```python
configfile: 'config.yml'

ALL_SUBWFS = config['subworkflows']

# <omitted> Check for min version 6.0

for MODULE in ALL_SUBWFS:

    module MODULE:
        snakefile:
            f'subworkflows/{MODULE}/Snakefile'

    use rule * from MODULE as f'{MODULE}_*'
```

```yml
# config.yml
subworkflows:
  - wf1
  - wf2
```

```bash
Snakefile
config.yml
└── subworkflows/
    ├── wf1/
    │   ├── Snakefile
    │   └── config.yml
    └── wf2/
        ├── Snakefile
        └── config.yml
```

**Describe the solution you'd like**

Given the above setup, I'd like 
1. to be able to run the root `Snakefile` which in turn runs the subworkflow, optionally in parallel when cores to do this are available,
2. each subworkflow to be able to use its own `config.yml` file.

**Describe alternatives you've considered**

Import each module explicitly, which is generally fine, but yields a lot of boilerplate.

**Additional context**

There is also my question about this on [StackOverflow](https://stackoverflow.com/questions/74731618/honour-modules-config-yml-when-dynamically-importing-modules-into-snakemake-w), which provides more details.

**Comment by pvandyken** [comment]
Closing, as this seems to be a duplicate of #1923.
